toolkit/tools/imageconfigvalidator/imageconfigvalidator.go

```diff
@@ -157,7 +157,7 @@ func validatePackages(config configuration.Config) (err error) {
 				return fmt.Errorf("%s: [ReadOnlyVerityRoot] and [TmpfsOverlayDebugEnabled] selected, but '%s' package is not included in the package lists", validateError, verityDebugPkgName)
 			}
 		}
-		if strings.Contains(kernelCmdLineString, fipsKernelCmdLine) {
+		if strings.Contains(kernelCmdLineString, fipsKernelCmdLine) || systemConfig.KernelCommandLine.EnableFIPS {
```

Comment:
Is adding `fips=1` to the kernel command line different from what happens when we set `EnableFIPS: true`? If not, I'm curious, if we need the new field.

Reply:
The `EnableFIPS: true` option operates differently in the particular case when the /boot partition is separate from the root partition. When the /boot partition is separate and `fips=1`, the kernel command line must contain a `boot=` parameter, which gets read during the fips self-tests during bootup to find the kernel's .hmac file which resides in the /boot partition.

So when `EnableFIPS: true`, the behavior adds the `boot=<reference to boot partition>` parameter (via UUID `boot=UUID=<boot partition's uuid>`) to the kernel cmdline in addition to `fips=1`.

If `fips=1` and `EnableFIPS: true` were both present, we would have two `fips=1` entries on the kernel cmdline and possibly the `boot=` parameter referenced above. And the kernel would only honor one `fips=1` setting (the last one set in the kernel cmdline).